lib/server.js

```diff
@@ -208,6 +209,29 @@ function listenPromise (server, listenOptions) {
   })
 }
 
+// This function used to compile a validation function for http version
+// bypass here is used to bypass the version check
+function compileValidateHTTPVersion (options) {
+  let bypass = false
+  const obj = {}
+  if (options.serverFactory) {
+    bypass = true
+  }
+  if (options.http2) {
+    obj['2.0'] = true
+    if (options.https && options.https.allowHTTP1 === true) {
+      obj['1.1'] = true
+      obj['1.0'] = true
+    }
+  } else {
+    obj['1.1'] = true
+    obj['1.0'] = true
+  }
+  return function validateHTTPVersion (httpVersion) {
```

Comment:
Can you add a comment explaining how this works?

Reply:
I have explain the function here in detail.
And added a document for the user how the expected behavior should be.